cmd/maya-apiserver/app/command/start.go

```diff
@@ -287,6 +287,13 @@ func (c *CmdStartOptions) setupMayaServer(mconfig *config.MayaConfig) error {
 
 	klog.Info("resources applied successfully by installer")
 
+	// clean older CASTemplates and RunTasks
+	cleanErr := install.SimpleInstaller().Clean()
```

Comment:
What is the reason for not exposing this as part of upgrade?
Upgrade has various version specific details. IMO cleanup can be done for all except last two or last version.
In other words, if upgrade is installing version _N_ then it might delete all the resources till version _N-2_ only if the current upgrade was successful. After the successful upgrade versions N-2, N-1 & N remain.

Reply:
Do we plan to expose dedicated controller to clean up such resources with specific intents?

Reply:
The clean up is done only after the upgrade is successful and the latest versions of CRs have been installed. IMO having the previous version CRs will not be helpful as if someone reverts the older version then installer will install the corresponding CRs automatically.

Reply:
ok.. There might be rare cases where user is using older CRs. Users might have scripted & so on.